phd-tests/framework/src/disk/file.rs

```diff
@@ -4,22 +4,84 @@
 
 //! Abstractions for disks with a raw file backend.
 
-use std::path::{Path, PathBuf};
-
+use camino::{Utf8Path, Utf8PathBuf};
 use propolis_client::types::{FileStorageBackend, StorageBackendV0};
-use tracing::{error, info};
+use tracing::{debug, error, warn};
 use uuid::Uuid;
 
-use crate::guest_os::GuestOsKind;
+use crate::{guest_os::GuestOsKind, zfs::ClonedFile as ZfsClonedFile};
+
+/// Describes the method used to create the backing file for a file-backed disk.
+#[derive(Debug)]
+enum BackingFile {
+    /// The disk is a ZFS clone of the original artifact.
+    Zfs(ZfsClonedFile),
+
+    /// The disk is a hard copy of the original artifact.
+    HardCopy(Utf8PathBuf),
+}
+
+impl BackingFile {
+    /// Creates a new backing file from the artifact with the supplied
+    /// `artifact_path`. If possible, this routine will create a ZFS clone of
+    /// the dataset containing the file; otherwise it will fall back to creating
+    /// a hard copy of the original artifact.
```

Comment:
the first thing i was looking for while reviewing was whether we fall back to not making ZFS clones if it doesn't work --- i'm glad to see that we do that!